### ISSUE TITLE
Adding Audio Generation?

### ISSUE THREAD

**Issue by frastlin** [goldfire/howler.js#754]
Hello,
I am wondering if it would be possible to add basic audio generation, probably as a plugin like spacial audio?
Noise:
https://noisehack.com/generate-noise-web-audio-api/
Sound:
http://webaudioapi.com/samples/oscillator/

It would also be nice if there was a library of generated sounds:
http://webaudioapi.com/samples/procedural/

Then if we could use the positional ability and panning ability of howler, it would be amazing!
Thanks,

**Comment by naugtur** [comment]
good idea for an independent library. The only interface required is being able to pass the generated source into a howler instance. I can help you scaffolding and publishing a library if you want to implement this.

**Comment by frastlin** [comment]
I don't understand the web audio API yet at all, which is why I'm using Howler... But if I can't figure out tone.js, then I may need to start digging into the API.
is it possible to wrap tone.js in a howler instance? Then control the playing, rate, position, seek and volume with the howler sound?
Would this be a preferred approach, or would building something from scratch be better?

**Comment by naugtur** [comment]
I'm not sure these two would work together out of the box. I don't see howler accepting an existing audio node to an instance. That'd probably have to happen first. Doesn't sound like an easy start though.
